business/dashboards.go

```diff
@@ -243,26 +341,98 @@ func (in *DashboardsService) buildRuntimesList(namespace string, templatesNames
 		if dashboard == nil {
 			continue
 		}
-		runtime := dashboard.Spec.Runtime
-		ref := models.DashboardRef{
-			Template: dashboard.Name,
-			Title:    dashboard.Spec.Title,
-		}
-		found := false
-		for i := range runtimes {
-			rtObj := &runtimes[i]
-			if rtObj.Name == runtime {
-				rtObj.DashboardRefs = append(rtObj.DashboardRefs, ref)
-				found = true
-				break
+		runtimes = addDashboardToRuntimes(dashboard, runtimes)
+	}
+	return runtimes
+}
+
+func (in *DashboardsService) fetchMetricNames(namespace, app, version string) []string {
+	labels := fmt.Sprintf(`{namespace="%s",app="%s"`, namespace, app)
```

Comment:
Probably it's documented somewhere.
Runtime metrics follow this format, or it is proposed to follow, right ?
I mean "app" here is the same "app" label with same meaning used in Istio, right ?
I wonder if that's case, perhaps we can use the "app" constant from istio config ? 
Just in case if that changes, or at least, in the code to try to not code "app" but use that constant.

But it sounds a cornercase, so if it's unlikely just disregard my comment, that can be done in a future PR.

Reply:
Actually, you're pointing me to something i didn't take care about: here the "app" label comes from prometheus label-mapping of k8s labels, so I should not hardcode "app" but use the same configurable App label that we use to identify k8s apps.

The same issue probably exists in the current code for custom dashboards.

Reply:
If you don't mind I will enter a JIRA to check and fix this potential issue here and in other places

Reply:
https://issues.jboss.org/browse/KIALI-2674

Reply:
(actually, I've fixed it here)